### PR TITLE
Device Attestation Information

### PR DESCRIPTION
**Pull request by maraino**
### Description

This PR adds support for device attestation SANs `Permanent Identifier` and `Hardware Module Name` as well as [Yubico PIV attestation extensions](https://developers.yubico.com/PIV/Introduction/PIV_attestation.html).

### CODE REVIEW

File: certinfo.go [hunk 1]
```diff
@@ -77,6 +82,61 @@ type stepCertificateAuthority struct {
...
+	TypeID asn1.ObjectIdentifier
+	Value  asn1.RawValue
+}
+
+// permanentIdentifier is defined in RFC 4043 as an optional feature that
+// may be used by a CA to indicate that two or more certificates relate to the
+// same entity.
+//
+// In device attestation this SAN will contain the UDID (Unique Device
+// IDentifier) or serial number of the device.
```
Comment (hslatman) [inline certinfo.go:100]: Unsure if this should be mentioned in this context.

File: yubico.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	if len(v) == 0 {
+		return []string{"unknown"}
+	}
+	policies := make([]string, 0, 2)
+	for i, b := range v {
+		switch i {
+		case 0:
+			switch b {
+			case 1:
+				policies = append(policies, "Pin policy: never")
```
Comment (hslatman) [inline yubico.go:61]:
```suggestion
				policies = append(policies, "PIN policy: never")
```
for all usages of PIN

File: yubico.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	return policies
+}
+
+func yubicoFormfactor(v []byte) string {
+	if len(v) == 0 {
+		return "unknown"
+	}
+	switch v[0] {
+	case 1, 81:
+		return "USB-A Keychain"
```
Comment (hslatman) [inline yubico.go:93]: Might be nice to have a `(FIPS)`-suffix for the `8x` ones?

Comment (maraino) [reply yubico.go:93]: I don't know, but it makes sense to add the suffix

Comment (maraino) [reply yubico.go]: Fixed with 416f2be

File: certinfo.go (hunk 1 above)
Comment (maraino) [reply certinfo.go]: Fixed with 1371e04

**Pull request merged**